src/main/java/org/jboss/modules/LocalModuleFinder.java

```diff
@@ -129,41 +129,37 @@ public LocalModuleFinder(boolean supportLayersAndAddOns) {
         return files;
     }
 
-    private static String toPathString(ModuleIdentifier moduleIdentifier) {
-        final StringBuilder builder = new StringBuilder(40);
-        builder.append(moduleIdentifier.getName().replace('.', File.separatorChar));
-        builder.append(File.separatorChar).append(moduleIdentifier.getSlot());
+    private static String toPathString(String moduleName) {
+        return moduleName.replace('.', File.separatorChar) + File.separatorChar + moduleName + File.separatorChar;
+    }
+
+    private static String toLegacyPathString(String moduleName) {
+        final ModuleIdentifier moduleIdentifier = ModuleIdentifier.fromString(moduleName);
+        final String name = moduleIdentifier.getName();
+        final String slot = moduleIdentifier.getSlot();
+        final StringBuilder builder = new StringBuilder(name.length() + slot.length() + 2);
+        builder.append(name.replace('.', File.separatorChar));
+        builder.append(File.separatorChar).append(slot);
         builder.append(File.separatorChar);
```

Comment:
In this case it doesn't matter if ending slash is present or not but it might be omitted too, doesn't it?